### PR TITLE
💻 Redesign of Adventure page within tryit endpoint

### PR DESCRIPTION
**Pull request by Annelein**
Fixes #5535 

https://www.figma.com/design/wrT20i7nKRBm66eVrf5A8F/Class-%2F%2F-Adventures-ideas?node-id=111-107&t=WR6z6PvgzdUO63sk-0

### CODE REVIEW

**Comment by Annelein** [comment]
A few good points that @AnneliesVlaar mentioned in #5517 , I've also added some

Few points of discussion:
- [ ] If the user uses the copy symbol, will the copy symbol (and button) than change into a paste symbol? And what behaviour will it have? Clean and paste? Paste at the end? Paste at the cursor?
- [ ] Do we want the explanation to be smaller in height and thus scrollable?

Missing in the design:
(Mentioned by @AnneliesVlaar in #5517)
- [ ] Suggestion box / cheatsheet / commands dropdown
- [ ] Custom name of the program
- [ ] Button to share the program
- [ ] Button to submit the program
- [ ] Button to read the program out loud
- [ ] #5204 

**Please leave a comment if there is something else you're missing or that needs to be discussed**

https://github.com/hedyorg/hedy/assets/48122190/4e42357b-c353-4bd3-b991-f47c8304821f

**Comment by Felienne** [comment]
> * If the user uses the copy symbol, will the copy symbol (and button) than change into a paste symbol? And what behaviour will it have? Clean and paste? 

Ow that is a great idea, I would love a copy button that changes into a paste button on theory. However in practice, no other tool has that (as fas as I know) so I fear it will be more confusing if we do it. We can disable the copy button if something has already been copied? But I also see how we can, for now, just aline with other tools do with is nothing.

> Paste at the end? Paste at the cursor?

This is an unrelated interesting question! We have long discussed pasting at the end instead of over for the yellow example buttons. I am fine with that but maybe it is out of scope here? Do we need to discuss that now?

> * [ ]  Do we want the explanation to be smaller in height and thus scrollable?

Another very hard question!!!! I think it does not look nice that the explanation "hops" when switching tabs. However kids might not care so much? I fear that they will not scroll and then miss important explanation if we don't show it all (but on the other other hand, maybe they never read?)

In conclusion I think I'd keep it as we have it, but I can easily be swayed.

**Comment by AnneliesVlaar** [comment]
> > * If the user uses the copy symbol, will the copy symbol (and button) than change into a paste symbol? And what behaviour will it have? Clean and paste?
> 
> Ow that is a great idea, I would love a copy button that changes into a paste button on theory. However in practice, no other tool has that (as fas as I know) so I fear it will be more confusing if we do it. We can disable the copy button if something has already been copied? But I also see how we can, for now, just aline with other tools do with is nothing.

I can see that an all-in-one copy/paste button can be confusing. I was wondering, if you copy your code (to use it in a different level/adventure I guess), how do you paste it there without a paste button. Can we easily teach student to use a button to copy and `ctrl+v` for example to paste?

> 
> > Paste at the end? Paste at the cursor?
> 
> This is an unrelated interesting question! We have long discussed pasting at the end instead of over for the yellow example buttons. I am fine with that but maybe it is out of scope here? Do we need to discuss that now?

I will make separate issue about the yellow button.

> 
> > * [ ]   Do we want the explanation to be smaller in height and thus scrollable?
> 
> Another very hard question!!!! I think it does not look nice that the explanation "hops" when switching tabs. However kids might not care so much? I fear that they will not scroll and then miss important explanation if we don't show it all (but on the other other hand, maybe they never read?)
> 
> In conclusion I think I'd keep it as we have it, but I can easily be swayed.

**Comment by gisellandrade** [comment]
@Annelein Here's the new buttons for the design and new modifications 
image

And also I just add the language buttons that shows this
Captura de pantalla 2024-07-09 093735

**Comment by Annelein** [comment]
@jpelay @Felienne just to be clear:
- Can I remove the tutorial button, see: https://discord.com/channels/978580085491785769/1177502764738281553/1262706361519833168
- The developers toggle in the adventure page is removed in this PR, what do we want with the 'Mandatory developers mode' toggle in customize class? Should we remove that option?

File: templates/public-adventures/body.html [hunk 1]
```diff
@@ -18,6 +17,39 @@
                 >
         </div>
         <div class="flex w-full sm:basis-1/2 gap-2">
+          <div id="adventures" class="flex-1">
```
Comment (jpelay) [inline templates/public-adventures/body.html:20]: Hi, Annelein! I don't really like the idea of the public adventures being visualized one by one only, and accessible through a filter. If we are removing the tabs (good idea), we should figure out an alternative way to access the adventures. No one is really going to find anything with this UI, because I dont think people will know the adventure's name in advance.

Also I think this change should be made in a sepearate PR, I think this one should tackle just the Hedy page.

File: static/js/cm-monokai-theme.ts [hunk 2]
```diff
@@ -13,10 +13,10 @@ const strawberry = "#ff6188",
     stone = "#7d8799", 
     malibu = "#61afef",
     violet = "#ae81ff",
-    background = "#272822",
+    background = "#0F1E33",
     highlightBackground = "#202020",
     selection = "#90cdf463",
-    gutterBackground = "#2F3129",
+    gutterBackground = "#0F1E33",
```
Comment (jpelay) [inline static/js/cm-monokai-theme.ts:19]: In my very personal and very biased opinion, I liked it more the way it was 😆 I can give an objective argument: I think the contrast between the keywords and the background looked better with the original theme, after all the theme wasn't designed by us, but rather by other people and we are just using it.

**Review by jpelay — COMMENTED** [review]
I like the redesign, and the UX, but I feel like we are overcorrecting going from a very colorful site, to a rather monocrhomatic one.

image

**Comment by jpelay** [comment]
For example the For Teachers page has more pop than this page, and this one is meant for kids.

image

**Comment by jpelay** [comment]
> @jpelay @Felienne just to be clear:
> 
> * Can I remove the tutorial button, see: https://discord.com/channels/978580085491785769/1177502764738281553/1262706361519833168
> * The developers toggle in the adventure page is removed in this PR, what do we want with the 'Mandatory developers mode' toggle in customize class? Should we remove that option?

* I think we can update the tutorial rather than remove it?
* Yes, I think we can discuss that tomorrow. I dont think it should be removed, it's being used a lot!

**Comment by Annelein** [comment]
> I like the redesign, and the UX, but I feel like we are overcorrecting going from a very colorful site, to a rather monocrhomatic one.
> 
> image

I understand what you mean and I agree! Only thing is, I have today and tomorrow left to work on this and we agreed on this design long ago so not sure what you want me to do with it for now.... Maybe @gisellandrade can make an updated design with a bit more colours which can easily be implemented in a different PR?

**Comment by jpelay** [comment]
> @jpelay I wasn't there on Tuesday during the meeting. But I will leave the "enforced developers mode" option.
> 
> One more question, the cheatsheet, language and read outloud dropdowns should be visible. But which of these things do we want to be visible in "enforced developers mode":
> 
> * level dropdown
> * adventure dropdown
> * next adventure button
> * program name input field
> * share button
> * submit button
> * progress bar

I think it could be similar to this, just hiding the adventure and making the editor bigger:

image

**Comment by jpelay** [comment]
What do we think about moving the "Read outloud" button down here?

image

**Comment by AnneliesVlaar** [comment]
> What do we think about moving the "Read outloud" button down here?
> 
> image

The "read outloud" functionality reads out loud the output and not the input, maybe it is better under/in the output screen?

**Comment by jpelay** [comment]
I've been thinking about this design, and I rearranged things a bit, in a way that I think improves usability of the page:

proposal_hedy

The changes I propose are the following:

1. Move the Read aloud button to be bellow the output window (like @AnneliesVlaar suggested)
2. Place the progress bar in the bottom of the page, as well as the next adventure button. It is more natural to go from top left to top right.
3. Replace the percentage with the number of adventures completed, and the number left. I think this can be more understandable by students who might not be familiar with percentages.
4. Move the code related buttons away from the editor.
5. Remove the text box with the title of the program. I think it's better if by clicking share, a modal is shown that prompts the student to title the program, and make it public or private. There might some badge in the UI showing that it was shared.
6. Make the adventure dropdown bigger, so it serves as a title to the current adventure. Another option would be to have a button  and open the adventure list to the side, like the folks at Code Academy do:

image

**Comment by gisellandrade** [comment]
I'm leaving here the drawer, and how it would look like 
image
image
image

**Comment by jpelay** [comment]
> As discussed in the meeting, if @boryanagoncharenko does a sanity check and merge, @MarleenGilsing can test it on alpha!

Yes! The url to try the new design is: `localhost:8080/tryit`

**Review by boryanagoncharenko — APPROVED** [review]
It is a big PR, did my best to go through the code! LGTM

**Comment by jpelay** [comment]
> It is a big PR, did my best to go through the code! LGTM

Yes, it's huuuuuge. Ill do my best to test thoroughly on Alpha!

**Comment by MarleenGilsing** [comment]
As I said I before, I think the redesign is great and looks a lot better and more professional. Great job! 

I tested it. These are the things I found: 
* The clear button is gonna create some very angry students who accidentally removed all their work. Solutions: Can we have an 'are you sure you want to remove all your code' pop-up? An 'undo' button would work too, or ctrl+Z? Or, last option: Do we even need the clear button? How many times does a student want to clear the whole field, and even if they do, it's not much effort to do so without having a clear button...?
* We need Dutch translations for some new buttons. 
* The button says 'show editor' while I think we have been using the term 'input screen'. I think editor might be better though, so maybe it doesn't need to change? 
* On my laptop the page looks like this:
image
There are nearly no margins at the sides (maybe because I don't have a very big screen) which makes it look a bit crowded. Can we add some margins? Same goes for the text of the adventures, I think it would look better if tehre was some white space on the sides of the text field. 
* I don't really like the underling of 'level 1 - introduction', I think it might look better without the underlining, but that might just be my taste. The underlining also makes it feel clickable, which it is not. 
* I really love the 'see more adventures' menu! I do think it's slightly counter intuitive that the button is on the right, while the menu appears on the left. 
* Extra idea for the 'see more adventures' menu: Would it be possible to show in the menu which of the adventures you have completed?  Like with a little tickmark or something behind the adventure?  
* (low priority) why is the 'e' in our font a smaller size? It looks kind of strange?
* (low priority) Before you could see if the adventure was just another adventure, or if you were learning a new command because of the tabs colours. We don't really have that anymore, exept for in the menu, but kids probably won't use that much. Maybe we can think of another way to show the students that they will learn a new command in this adventure. 
* In the parsons problem it's not really clear that you can do 2 exercises and that you have to click the 2. (Same as the old design though). 
* BUG: If you finish the level 1 quiz and click 'go to level 2' you'll be sent to the quiz of level 2, instead of the introduction.

**Comment by Felienne** [comment]
Thanks for the superextensive review Marleen!!! I marked the PR tobediscussed again so we can go over the issues in next week's meeting

**Comment by jpelay** [comment]
> As I said I before, I think the redesign is great and looks a lot better and more professional. Great job!
> 
> I tested it. These are the things I found:
> 
> * The clear button is gonna create some very angry students who accidentally removed all their work. Solutions: Can we have an 'are you sure you want to remove all your code' pop-up? An 'undo' button would work too, or ctrl+Z? Or, last option: Do we even need the clear button? How many times does a student want to clear the whole field, and even if they do, it's not much effort to do so without having a clear button...?
> * We need Dutch translations for some new buttons.
> * The button says 'show editor' while I think we have been using the term 'input screen'. I think editor might be better though, so maybe it doesn't need to change?
> * On my laptop the page looks like this:
>   image
>   There are nearly no margins at the sides (maybe because I don't have a very big screen) which makes it look a bit crowded. Can we add some margins? Same goes for the text of the adventures, I think it would look better if tehre was some white space on the sides of the text field.
> * I don't really like the underling of 'level 1 - introduction', I think it might look better without the underlining, but that might just be my taste. The underlining also makes it feel clickable, which it is not.
> * I really love the 'see more adventures' menu! I do think it's slightly counter intuitive that the button is on the right, while the menu appears on the left.
> * Extra idea for the 'see more adventures' menu: Would it be possible to show in the menu which of the adventures you have completed?  Like with a little tickmark or something behind the adventure?
> * (low priority) why is the 'e' in our font a smaller size? It looks kind of strange?
> * (low priority) Before you could see if the adventure was just another adventure, or if you were learning a new command because of the tabs colours. We don't really have that anymore, exept for in the menu, but kids probably won't use that much. Maybe we can think of another way to show the students that they will learn a new command in this adventure.
> * In the parsons problem it's not really clear that you can do 2 exercises and that you have to click the 2. (Same as the old design though).
> * BUG: If you finish the level 1 quiz and click 'go to level 2' you'll be sent to the quiz of level 2, instead of the introduction.

Thank you sooo much for the feedback!!! May I ask what's your screen resolution?

**Comment by MarleenGilsing** [comment]
You're welcome! My screen resolution is 1536x864
